pilot/pkg/config/memory/store.go

```diff
@@ -147,7 +147,7 @@ func (cr *store) Create(cfg config.Config) (string, error) {
 	kind := cfg.GroupVersionKind
 	s, ok := cr.schemas.FindByGroupVersionKind(kind)
 	if !ok {
-		return "", fmt.Errorf("unknown type %v", kind)
```

Comment:
this loses context and makes it hard to understand why it failed

Reply:
Please don't remove this. It is very useful to detect unexpected kind resource when we try to store it.

Reply:
ok